src/ecs/rendering/resources.rs

```diff
@@ -167,3 +167,32 @@ impl WindowMessages {
         self.queue.push(Box::new(command));
     }
 }
+
+/// World resource that stores screen dimensions.
+pub struct ScreenDimensions {
+    /// Screen width in pixels (px).
+    pub w: f32,
+    /// Screen height in pixels (px).
+    pub h: f32,
+    /// Width divided by height.
+    pub aspect_ratio: f32,
+}
```

Comment:
A `dirty` flag maybe? So the user doesn't have to cache the size for change detection. The `RenderSystem` can clear the dirty flag just before doing computation.